PowerPointLabs/PowerPointLabs/CropLab/CropToSame.cs

```diff
@@ -0,0 +1,169 @@
+                    float cropBottom = Math.Max(shapes[1].PictureFormat.CropBottom, epsilon);
+                    float cropLeft = Math.Max(shapes[1].PictureFormat.CropLeft, epsilon);
+                    float cropRight = Math.Max(shapes[1].PictureFormat.CropRight, epsilon);
+
+                    float refShapeCroppedHeight = cropTop + cropBottom;
+                    float refShapeCroppedWidth = cropLeft + cropRight;
+                    
+                    shapes[i].PictureFormat.CropTop = Math.Max(0, heightToCrop * cropTop / refShapeCroppedHeight / scaleHeight);
+                    shapes[i].PictureFormat.CropLeft = Math.Max(0, widthToCrop * cropLeft / refShapeCroppedWidth / scaleWidth);
+                    shapes[i].PictureFormat.CropRight = Math.Max(0, widthToCrop * cropRight / refShapeCroppedWidth / scaleWidth);
+                    shapes[i].PictureFormat.CropBottom = Math.Max(0, heightToCrop * cropBottom / refShapeCroppedHeight / scaleHeight);
+                    
+                }
+            }
+            catch (Exception e)
+            {
+                if (handleError)
+                {
+                    ProcessErrorMessage(e);
+                    return;
+                }
+
+                throw;
+            }
+            
+        }
+
+        private static bool VerifyIsShapeRangeValid(PowerPoint.ShapeRange shapeRange, bool handleError)
+        {
+            try
+            {
+                if (shapeRange.Count < 1)
+                {
+                    ThrowErrorCode(ErrorCodeForSelectionCountZero);
+                }
+
+                if (!IsPictureForSelection(shapeRange))
+                {
+                    ThrowErrorCode(ErrorCodeForSelectionNonPicture);
+                }
+
+                return true;
+            }
+            catch (Exception e)
+            {
+                if (handleError)
+                {
+                    ProcessErrorMessage(e);
+                    return false;
+                }
+
+                throw;
+            }
+        }
+
+        private static void VerifyIsSelectionValid(PowerPoint.Selection selection)
+        {
+            if (selection.Type != PowerPoint.PpSelectionType.ppSelectionShapes)
+            {
+                ThrowErrorCode(ErrorCodeForSelectionCountZero);
+            }
+        }
+
+        private static bool IsPictureForSelection(PowerPoint.ShapeRange shapeRange)
+        {
+            return (from PowerPoint.Shape shape in shapeRange select shape).All(IsPicture);
+        }
+
+        private static bool IsPicture(PowerPoint.Shape shape)
+        {
+            return shape.Type == Office.MsoShapeType.msoPicture ||
+                   shape.Type == Office.MsoShapeType.msoLinkedPicture;
+        }
+
+        private static void ThrowErrorCode(int typeOfError)
+        {
+            throw new Exception(typeOfError.ToString(CultureInfo.InvariantCulture));
```

Comment:
define a specific exception? like CropLabOperationException something like that, instead of using the generic one